### PR TITLE
[stable/nginx-ingress] tidy up updateStrategy

### PR DESCRIPTION
**Pull request by egeland**
* Add `minReadySeconds` to help reduce outages

* Add missing `updateStrategy` block in daemonset config

### CODE REVIEW

File: stable/nginx-ingress/values.yaml [hunk 1]
```diff
@@ -65,6 +65,11 @@ controller:
   #    maxUnavailable: 1
   #  type: RollingUpdate
 
+  # minReadySeconds to avoid killing pods before we are ready
+  ##
+  minReadySeconds: 15
```
Comment (unguiculus) [inline stable/nginx-ingress/values.yaml]: Isn't that kind of high as a default?

**Review by unguiculus — CHANGES_REQUESTED** [review]
Please fix conflicts.

File: stable/nginx-ingress/values.yaml (hunk 1 above)
Comment (egeland) [reply stable/nginx-ingress/values.yaml]: We've been using this in production for a while - it may be ok to lower it. What would you suggest?

Comment (unguiculus) [reply stable/nginx-ingress/values.yaml]: The default is zero. I think we should stick to that.

**Comment by egeland** [comment]
@unguiculus I set the default `minReadySeconds` to `0` as suggested.

File: stable/nginx-ingress/values.yaml (hunk 1 above)
Comment (egeland) [reply stable/nginx-ingress/values.yaml]: Fixed 👍

**Comment by egeland** [comment]
Anything further to do on this, @unguiculus ? Would be really great to get it merged in. 😄

File: stable/nginx-ingress/README.md [hunk 2]
```diff
@@ -60,6 +60,7 @@ Parameter | Description | Default
 `controller.extraArgs` | Additional controller container arguments | `{}`
 `controller.kind` | install as Deployment or DaemonSet | `Deployment`
 `controller.tolerations` | node taints to tolerate (requires Kubernetes >=1.6) | `[]`
+`controller.minReadySeconds` | how many seconds a pod needs to be ready before killing the next, during update | `15`
```
Comment (mattfarina) [inline stable/nginx-ingress/README.md]: Could you please update the readme for the time changes as well?

**Review by mattfarina — COMMENTED** [review]
/ok-to-test

File: stable/nginx-ingress/README.md (hunk 2 above)
Comment (egeland) [reply stable/nginx-ingress/README.md]: Done 👍

**Comment by egeland** [comment]
Hmmm... my rebase caused some duplication of commits - should disappear once this is merged/squashed...

**Pull request merged**